sets/maps.go

```diff
@@ -44,8 +44,14 @@ func IntersectMapKeys[K comparable, V any](maps ...map[K]V) map[K]V {
 	}
 	smallestMap := maps[smallestIdx]
 	final := make(map[K]V, len(smallestMap))
```

Comment:
I don't know if preallocating final with `len(smallestMap)`  is the right tradeoff or not. You are saving at most log(n) allocs and n copies, but possibly wasting n space in the scenario that the smallest map is large but the intersection is small. (Dave pointed this out in a discussion)